runtimes/server-interface/workspace.ts

```diff
@@ -16,6 +16,7 @@ interface Dirent {
  */
 export type Workspace = {
     getTextDocument: (uri: string) => Promise<TextDocument | undefined>
+    getAllTextDocuments: () => TextDocument[]
```

Comment:
Is there a reason servers can't watch `https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_didOpen` and keep track of the interesting ones themselves?

Reply:
They of course can, but I think it's not scalable approach. It pushes unnecessary complexity to each Server implementation - if tomorrow server developer decides they need access files, they need to reimplement caching and monitoring. Here, this API is provided from central place, which keeps track of open documents anyways.

Reply:
I think I'd rather convert `getAllTextDocuments` to `async`, for future extensibility and to keep it consistent with rest of methods in Workspace feature.

Reply:
> I think I'd rather convert `getAllTextDocuments` to `async`, for future extensibility and to keep it consistent with rest of methods in Workspace feature.

I agree with this approach